src/components/hosts/EditHostForm.tsx

```diff
@@ -42,7 +42,10 @@ type EditHostFormProps = {
 const validationSchema = (initialValues: HostUpdateParams, hosts: Host[] = []) =>
   Yup.object().shape({
     hostname: hostnameValidationSchema.concat(
-      uniqueHostnameValidationSchema(initialValues.hostname, hosts),
+      uniqueHostnameValidationSchema(initialValues.hostname, hosts).notOneOf(
+        ['localhost'],
```

Comment:
This does not handle `localhost.localdomain` but backend validation should be enhanced as well. I wrote a comment to BZ mentioning it.